extensions/functions/src/main/java/io/crate/window/OffsetValueFunctions.java

```diff
@@ -89,29 +183,40 @@ public Signature boundSignature() {
     public Object execute(int idxInPartition,
                           WindowFrameState currentFrame,
                           List<? extends CollectExpression<Row, ?>> expressions,
-                          Input... args) {
+                          Boolean ignoreNulls,
+                          Input[] args) {
+        boolean ignoreNullsOrFalse = ignoreNulls != null && ignoreNulls;
```

Comment:
I wonder if we should change the signatures. `ignore nulls` is static during the life-cycle of the window function execution, so we could already pass the right value when the function instance is constructed and remove the parameter from `execute`

That would have the advantage that we could have two implementations for the various functions and remove a lot of `if`s within the `execute` path.  `execute` is a hot-spot and called per row and I think it would be worth to optimize this further.


Sorry for bringing this up so late in the review-cycle, but it didn't occur to me earlier.

Reply:
No problem. As discussed from the previous review comment, it would be 3 implementations instead of 2, since I need `Boolean` instead of `boolean`. Also, it would optimize away 1~2 branching per row. I guess you mean a lot of `if`s because of the number of branching is linearly dependent on the number of rows? I have started to handle this but want to re-confirm.